### PR TITLE
added R Skeleton and 2 functions

### PR DESCRIPTION
**Pull request by ofelyaaa**

### CODE REVIEW

File: resmaswrapped/wrapped.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+pip install spotipy
```
Comment (Ilonka-v-Vliet) [inline resmaswrapped/wrapped.py]: should the pip install be in this file?

File: resmaswrapped/wrapped.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import requests
+import spotipy
+from spotipy.oauth2 import SpotifyOAuth
+import pandas as pd
+import time
+
+SPOTIPY_CLIENT_ID='4e6464520c2b4cf3a90ad792e5ad6504'
+SPOTIPY_CLIENT_SECRET='4dde9d4fbb604ad3b22c0ad7204e25a0'
```
Comment (Ilonka-v-Vliet) [inline resmaswrapped/wrapped.py]: I think these codes might be account-specific, and also allow you to log in to someone's account? spotipy's documentation refers to them as credentials. If they are, you should probably delete them from github and github's history, unless you were using a throwaway account where it doesn't matter if someone can access it

File: resmaswrapped/wrapped.py
Comment (Ilonka-v-Vliet) [inline resmaswrapped/wrapped.py:1]: overall, I'm not sure what the code is meant to do and so I can't judge whether it's doing what it needs to do. It seems like it's meant to collect information of the recent most-listened-to tracks of an account on spotify? All the code for that looks sensible. For a package though, you should make sure everything that happens in your script can be done by calling a function (I haven't done this yet either for my project). Maybe the lines that aren't already in a function can be gathered into something like CreateTrackDatabase() that calls on the other functions?

File: resmaswrapped/setup.py
Comment (Ilonka-v-Vliet) [inline resmaswrapped/setup.py:1]: Looks good!

File: resmaswrapped/wrapped.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ artist = meta['album']['artists'][0]['name']
+ spotify_url = meta['external_urls']['spotify']
+ album_cover = meta['album']['images'][0]['url']
+ track_info = [name, album, artist, spotify_url, album_cover]
+ return track_info
+
+# loop over track ids
+tracks = []
+for i in range(len(track_ids)):
+ time.sleep(.5)
```
Comment (Ilonka-v-Vliet) [inline resmaswrapped/wrapped.py]: Is there a reason for the time.sleep()? If so, it would be good to mention in a comment

**Review by Ilonka-v-Vliet — APPROVED** [review]
Overall it looks like the code you wrote should work, but the documentation is missing which makes it hard to tell if it's working as intended. It also seems to work as a script to run, rather than as a package to use functions from. Lastly, I hope the client_id and client_secret values are not as sensitive as I think they are, otherwise you should probably delete them from the github and replace them with a function that can ask for the user's credentials.

File: resmaswrapped/wrapped.py
Comment (ofelyaaa) [reply resmaswrapped/wrapped.py:1]: Thank you so much for your feedback! It made me realise that I accidentally uploaded earlier versions of the code to github, my new pull request should include documentation and comments, sorry for the confusion!

File: resmaswrapped/OLD wrapped.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import requests
+import spotipy
+from spotipy.oauth2 import SpotifyOAuth
+import pandas as pd
+import time
+
+SPOTIPY_CLIENT_ID='insert client id'
+SPOTIPY_CLIENT_SECRET='insert client secret'
+SPOTIPY_REDIRECT_URI='http://localhost:8888/callback'
+SCOPE = 'user-top-read'
```
Comment (Bamske) [inline resmaswrapped/OLD wrapped.py:11]: It looks like the user needs to write this up manually, would it be easier to have them input this into a funtion?

**Review by Bamske — COMMENTED** [review]
I looked at the new script you updated. Like Ilonka mentioned it is now completely clear what your package is for so first of it would be great to add that to the README file which is currently empty.
I see that you added some documentation which I think could be expanded on somewhat with usage examples, and maybe you could clarify which functions are supposed to be called on by the user and which not (this might also be more clear once you update the README)

**Pull request merged**